### PR TITLE
[Fix](nereids) fix NormalizeRepeat, change the outputExpression rewrite logic

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
In NormalizeRepeat, three parts of the outputExpression of LogicalRepeat need to be pushed down and outputted by bottom project: flattenGroupingSetExpr, argumentsOfGroupingScalarFunction, argumentsOfAggregateFunction.
In the original code, use these three parts to rewrite the outputExpressions of LogicalRepeat to slots.This can cause problems in some cases, for example:
```sql
SELECT
	ROUND( SUM(pk + 1) - 3) col_alias1,
	pk + 1 AS col_alias3 
	FROM
	table_20_undef_partitions2_keys3_properties4_distributed_by53
GROUP BY
	GROUPING SETS ((pk), ()) ;
```
The three parts expression needed to be pushed down are: pk, pk+1. The original code use pk+1 to rewrite the pk + 1 AS col_alias3  to slot. But the pk+1 is not in the list of grouping outputs, and then report error.
This pr change the rewrite process,  divide the expression needed to be pushed down  into 2 parts: one is (flattenGroupingSetExpr) and the other one is (argumentsOfGroupingScalarFunction, argumentsOfAggregateFunction).
 and use the flattenGroupingSetExpr rewrite all LogicalRepeat outputExpressions, and use the argumentsOfGroupingScalarFunction, argumentsOfAggregateFunction to rewrite only the agg function arguments and the grouping scalar function.
So, in the above sql, the pk + 1 AS col_alias3  will not be rewritten to slot, and can be computed.

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run external

**Comment by feiniaofeiafei** [comment]
run cloud_p1

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeRepeat.java [hunk 1]
```diff
@@ -117,19 +117,29 @@ private void checkGroupingSetsSize(LogicalRepeat<Plan> repeat) {
...
 
-        // replace the arguments of grouping scalar function to virtual slots
-        // replace some complex expression to slot, e.g. `a + 1`
-        List<NamedExpression> normalizedAggOutput = context.normalizeToUseSlotRef(
-                        repeat.getOutputExpressions(), this::normalizeGroupingScalarFunction);
+        // use argsContext
+        // rewrite the arguments of grouping scalar function to slots
+        // rewrite grouping scalar function to virtual slots
+        // rewrite the arguments of agg function to slots
+        List<NamedExpression> normalizedAggOutput = repeat.getOutputExpressions().stream()
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/NormalizeRepeat.java:137]: could we replace all stream api by for loop?

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run p0

**Comment by feiniaofeiafei** [comment]
run cloud_p1

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]

**Pull request merged**